### PR TITLE
[TECHNICAL SUPPORT] LPS-106352

### PR DESCRIPTION
**Pull request by leoadb**
https://issues.liferay.com/browse/LPS-106352

Asset Publisher filter by field doesn't work with multiple selection fields, because only first element is used.

I am returning a String[] in case select field has more than one selected element and adding one query term for each array element.

/cc: @dacousalr @ealonso @adolfopa

### CODE REVIEW

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMImpl.java [hunk 1]
```diff
@@ -486,7 +487,12 @@ else if (type.equals(DDMImpl.TYPE_SELECT)) {
 
 			String[] stringArray = ArrayUtil.toStringArray(jsonArray);
 
-			fieldValue = stringArray[0];
+			if (stringArray.length > 1) {
+				fieldValue = stringArray;
+			}
+			else {
+				fieldValue = stringArray[0];
```
Comment (jeyvison) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMImpl.java]: Hey @jorgediaz-lr . I believe this line can throw an `ArrayIndexOutOfBoundsException`  if `lenght == 0`. Am I right? Can this lenght be 0?

**Comment by jorgediaz-lr** [comment]
Hi @jeyvison

I don't know if length can be zero or not.

Nevertheless, old code was:
fieldValue = stringArray[0];

So in case length == 0, old code should be fail as well.

My idea was to leave old behavior in case `stringArray.length <= 1` to avoid any other unexpected behavior change and modify `stringArray.length>1` case.

**Comment by jeyvison** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by jeyvison** [comment]
Pull request submitted to https://github.com/brianchandotcom/liferay-portal/pull/83016. <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Pull request closed without merging**